### PR TITLE
bingo: init at 0.5.2

### PR DESCRIPTION
**Pull request by aaronjheng**
###### Motivation for this change

Like `go get` but for Go tools! CI Automating versioning of Go binaries in a nested, isolated Go modules.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/bingo/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    repo = "bingo";
+    rev = "v${version}";
+    sha256 = "sha256-4D8YaA/AH1gIp5iwD7WEAdBl73sqwHpfOe7bnxVcRcw=";
+  };
+
+  vendorSha256 = "sha256-xrz9FpwZd+FboVVTWSqGHRguGwrwE9cSFEEtulzbfDQ=";
+
+  patches = [
+    # Skip e2e tests.
+    ./e2e_test.patch
```
Comment (zowoq) [inline pkgs/development/tools/bingo/default.nix:18]: Probably easier to set `doCheck = false` than keeping a 1300 line patch around.

File: pkgs/development/tools/bingo/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    package = bingo;
+    command = "bingo version";
+    version = "v${version}";
+  };
+
+  meta = with lib; {
+    description = "Like `go get` but for Go tools! CI Automating versioning of Go binaries in a nested, isolated Go modules.";
+    homepage = "https://github.com/bwplotka/bingo";
+    license = licenses.asl20;
+    platforms = platforms.linux ++ platforms.darwin;
```
Comment (zowoq) [inline pkgs/development/tools/bingo/default.nix:37]: Default for `buildGoModule`.
```suggestion
```

File: pkgs/development/tools/bingo/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/tools/bingo/default.nix:18]: or do `rm` on the deleted file.

Comment (aaronjheng) [reply pkgs/development/tools/bingo/default.nix:18]: Thanks guys, this patch deletes two test files. `doCheck = false` sounds good, But I do not want to skip the check phase.

Comment (zowoq) [reply pkgs/development/tools/bingo/default.nix:18]: Don't need to patch them out, just delete them.

```nix
  postPatch = ''
    rm get_e2e_test.go get_e2e_utils_test.go
  '';
```

Comment (aaronjheng) [reply pkgs/development/tools/bingo/default.nix]: I see. Nice.

**Pull request merged**